erpnext/stock/doctype/stock_entry/stock_entry.js

```diff
@@ -323,6 +323,12 @@ frappe.ui.form.on('Stock Entry', {
 		attach_bom_items(frm.doc.bom_no)
 	},
 
+	before_save: function(frm) {
+		frm.doc.items.forEach((item) => {
+			item.uom = item.uom??item.stock_uom;
```

Comment:
```suggestion
			item.uom = item.uom || item.stock_uom;
```

Both are equivalent in this case. Switching to or for better compatibility.